cmd/kubelet/app/server.go

```diff
@@ -246,6 +247,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.MaxPods, "max-pods", 40, "Number of Pods that can run on this Kubelet.")
 	fs.StringVar(&s.DockerExecHandlerName, "docker-exec-handler", s.DockerExecHandlerName, "Handler to use when executing a command in a container. Valid values are 'native' and 'nsenter'. Defaults to 'native'.")
 	fs.StringVar(&s.PodCIDR, "pod-cidr", "", "The CIDR to use for pod IP addresses, only used in standalone mode.  In cluster mode, this is obtained from the master.")
+	fs.StringVar(&s.ResolverConfig, "resolver-config", kubelet.ResolvConfDefault, "Resolver configuration file used as the basis for the container DNS resolution configuration.")
```

Comment:
Will change the flag name to resolv-conf as suggested.

This is a quote from docker documentation:
"Regarding DNS settings, in the absence of either the --dns=IP_ADDRESS... or the --dns-search=DOMAIN... option, Docker makes each container’s /etc/resolv.conf look like the /etc/resolv.conf of the host machine"

If no options get specified to docker then the outcome is the configuration specified by /etc/resolv.conf. The kubelet --resolv-conf flag is intended to allow the user to override this behavior by specifying a different "Default DNS" setting... the cluster DNS setting is incremental on the "Default DNS" setting.

So from a behavior perspective, specifying --resolv-conf="" or --resolv-conf="/etc/resolv.conf" should yield a similar result... but the implementation could differ.

Matrix:


resolv.conf            |  DNS        | behavior                                     |
------------------------|---------------|-------------------------------------------
| ""                          | default       | docker default (/etc/resolv.conf) |
| /etc/resolv.conf     | default      | (1) current patch uses docker default
| user specified      | default       | pass resolv.conf (nameservers, search) to docker
| ""                          | cluster       | (2) clusterDNS only ? |
| /etc/resolv.conf    | cluster       | current behavior |
| user specified      | cluster       | file + clusterDNS |

Can you please provide input on (1) and (2). I believe the other cases are self evident.